examples/advanced_pytorch/utils.py

```diff
@@ -0,0 +1,130 @@
+import torch
+import torchvision.transforms as transforms
+from torchvision.datasets import CIFAR10
+
+import warnings
+
+warnings.filterwarnings("ignore")
+
+DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+
+def load_data():
+    """Load CIFAR-10 (training and test set)."""
+    transform = transforms.Compose(
+        [
+            transforms.Resize(256),
+            transforms.CenterCrop(224),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+        ]
+    )
+
+    trainset = CIFAR10("./dataset", train=True, download=True, transform=transform)
+    testset = CIFAR10("./dataset", train=False, download=True, transform=transform)
+
+    num_examples = {"trainset": len(trainset), "testset": len(testset)}
+    return trainset, testset, num_examples
+
+
+def load_partition(idx: int):
+    """Load 1/10th of the training and test data to simulate a partition."""
+    assert idx in range(10)
+    trainset, testset, num_examples = load_data()
+    n_train = int(num_examples["trainset"] / 10)
+    n_test = int(num_examples["testset"] / 10)
+
+    train_parition = torch.utils.data.Subset(
+        trainset, range(idx * n_train, (idx + 1) * n_train)
+    )
+    test_parition = torch.utils.data.Subset(
+        testset, range(idx * n_test, (idx + 1) * n_test)
+    )
+    return (train_parition, test_parition)
+
+
+def train(net, trainloader, valloader, epochs):
+    """Train the network on the training set."""
+    print("Starting training...")
+    net.to(DEVICE)  # move model to GPU if available
+    criterion = torch.nn.CrossEntropyLoss().to(DEVICE)
+    optimizer = torch.optim.SGD(
+        net.parameters(), lr=0.1, momentum=0.9, weight_decay=1e-4
+    )
+    net.train()
+    for _ in range(epochs):
+        for images, labels in trainloader:
+            images, labels = images.to(DEVICE), labels.to(DEVICE)
+            optimizer.zero_grad()
+            loss = criterion(net(images), labels)
+            loss.backward()
+            optimizer.step()
```

Comment:
Making sure `train` receives a device. 
```suggestion
def train(net, trainloader, valloader, epochs, device= "cpu"):
    """Train the network on the training set."""
    print("Starting training...")
    net.to(device)  # move model to GPU if available
    criterion = torch.nn.CrossEntropyLoss().to(device)
    optimizer = torch.optim.SGD(
        net.parameters(), lr=0.1, momentum=0.9, weight_decay=1e-4
    )
    net.train()
    for _ in range(epochs):
        for images, labels in trainloader:
            images, labels = images.to(device), labels.to(device)
            optimizer.zero_grad()
            loss = criterion(net(images), labels)
            loss.backward()
            optimizer.step()
```